ethereumj-core/src/test/java/org/ethereum/sharding/processing/consensus/ShufflingCommitteeFactoryTest.java

```diff
@@ -0,0 +1,167 @@
+package org.ethereum.sharding.processing.consensus;
+
+import org.ethereum.crypto.HashUtil;
+import org.ethereum.sharding.processing.state.Committee;
+import org.ethereum.util.FastByteComparisons;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.ethereum.sharding.processing.consensus.BeaconConstants.CYCLE_LENGTH;
+import static org.ethereum.sharding.processing.consensus.BeaconConstants.MIN_COMMITTEE_SIZE;
+import static org.ethereum.sharding.processing.consensus.BeaconConstants.SHARD_COUNT;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * @author Mikhail Kalinin
+ * @since 24.09.2018
+ */
+public class ShufflingCommitteeFactoryTest {
+
+    @Test
+    public void testCreate() {
+        int startShard = 3;
+        Committee[][] committees = new ShufflingCommitteeFactory().create(HashUtil.randomHash(),
+                IntStream.range(1, 3 * SHARD_COUNT * MIN_COMMITTEE_SIZE * CYCLE_LENGTH / 2).toArray(), startShard);
+
+        // each committee dedicated for own shard
+        int shard = startShard;
+        for (Committee[] slot : committees) {
+            for (Committee committee : slot) {
```

Comment:
slot.length == SHARD_COUNT ?

Reply:
nope, slot.length == CYCLE_LENGTH

Reply:
Sorry, `slot.length` depends on the size of active validator set. It varies from `1` to `(1 << 24) / CYCLE_LENGTH / (MIN_COMMITTEE_SIZE * 2) + 1`, it's calculated with [ShufflingCommitteeFactory#calcShardsPerSlot()](https://github.com/ethereum/ethereumj/blob/sharding/committees/ethereumj-core/src/main/java/org/ethereum/sharding/processing/consensus/ShufflingCommitteeFactory.java#L110)

Reply:
Got it